tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,225 @@
+					BeforeAll(func() {
+						istioYAML := `
+apiVersion: operator.istio.io/v1alpha1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: %s`
+						istioYAML = fmt.Sprintf(istioYAML, version, controlPlaneNamespace)
+						fmt.Printf("Istio CR YAML: %s\n", istioYAML)
+						Expect(kubectl.ApplyString(istioYAML)).
+							Should(Succeed(), "Istio CR failed to be created")
+						Success("Istio CR created")
+					})
+
+					It("updates the Istio resource status to Reconcilied and Ready", func() {
+						Eventually(kubectl.GetConditions).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReconcilied), "Istio is not Reconcilied; unexpected Condition")
+
+						Eventually(kubectl.GetConditions).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReady), "Istio is not Ready; unexpected Condition")
+
+						Success("Istio resource is Reconcilied and Ready")
+					})
+
+					It("deploys istiod", func() {
+						Expect(kubectl.GetDeployments(controlPlaneNamespace)).
+							To(Equal([]string{"istiod"}), "Istiod deployment is not present; unexpected list of deployments")
+						Success("Istiod deployment is present")
+
+						// TODO: we need to add a function to get the istio version from the control panel directly
+						// and compare it with the applied version
+						// This is a TODO because actual version.yaml contains for example latest and not the actual version
+						// Posible solution is to add actual version to the version.yaml
+					})
+
+					It("deploys the CNI DaemonSet when running on OpenShift", func() {
+						if ocp {
+							Eventually(kubectl.GetDaemonSets).
+								WithArguments(namespace).
+								Should(ContainElement("istio-cni-node"), "CNI DaemonSet is not deployed; expected list to contain element")
+
+							Expect(kubectl.GetPodPhase(namespace, "k8s-app=istio-cni-node")).Should(Equal("Running"), "CNI Daemon is not running; unexpected Phase")
+							Success("CNI DaemonSet is deployed in the namespace and Running")
+						} else {
+							Consistently(kubectl.GetDaemonSets).
+								WithArguments(namespace).WithTimeout(30*time.Second).
+								Should(BeEmpty(), "CNI DaemonSet is present; expected list to be empty")
+							Success("CNI DaemonSet is not deployed in the namespace because it's not OpenShift")
+						}
+					})
+
+					It("doesn't continuously reconcile the istio resource", func() {
+						Eventually(kubectl.Logs).
+							WithArguments(controlPlaneNamespace, "app=istiod", 30*time.Second).
+							ShouldNot(ContainSubstring("Reconciliation done"), "Istio Operator is continuously reconciling")
+						Success("Istio Operator stopped reconciling")
+					})
+				})
+
+				When("the Istio CR is deleted", func() {
+					BeforeEach(func() {
+						Expect(kubectl.Delete(controlPlaneNamespace, "istio", istioName)).To(Succeed(), "Istiod deployment failed to be deleted")
```

Comment:
Wrong failure message. We're not deleting the deployment. We're deleting the Istio CR.

Reply:
Fixed